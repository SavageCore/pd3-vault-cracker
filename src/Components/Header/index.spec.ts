import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import Component from './index.svelte';

test('should render without crashing', () => {
  const { container } = render(Component);
  expect(container).toMatchSnapshot();
});
