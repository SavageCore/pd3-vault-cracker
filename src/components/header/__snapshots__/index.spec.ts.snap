// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render without crashing 1`] = `
<div>
  <div
    class="grid header svelte-66di5t"
  >
    <div
      class="logo svelte-66di5t"
    >
      PD3 Vault Cracker
    </div>
     
    <div
      class="offlinemode svelte-66di5t"
    >
      #OFFLINEMODE
    </div>
     
    <div
      class="links svelte-66di5t"
    >
      <div
        aria-checked="false"
        aria-label="auto"
        aria-live="polite"
        class="icon theme-toggle svelte-66di5t"
        id="theme-toggle"
        role="checkbox"
        tabindex="0"
        title="Dark theme"
      >
        <svg
          height="2rem"
          viewBox="0 0 24 24"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.742 13.045a8.088 8.088 0 0 1-2.077.271c-2.135 0-4.14-.83-5.646-2.336a8.025 8.025 0 0 1-2.064-7.723A1 1 0 0 0 9.73 2.034a10.014 10.014 0 0 0-4.489 2.582c-3.898 3.898-3.898 10.243 0 14.143a9.937 9.937 0 0 0 7.072 2.93 9.93 9.93 0 0 0 7.07-2.929 10.007 10.007 0 0 0 2.583-4.491 1.001 1.001 0 0 0-1.224-1.224zm-2.772 4.301a7.947 7.947 0 0 1-5.656 2.343 7.953 7.953 0 0 1-5.658-2.344c-3.118-3.119-3.118-8.195 0-11.314a7.923 7.923 0 0 1 2.06-1.483 10.027 10.027 0 0 0 2.89 7.848 9.972 9.972 0 0 0 7.848 2.891 8.036 8.036 0 0 1-1.484 2.059z"
          />
        </svg>
      </div>
       
      <div
        class="icon pointer svelte-66di5t"
        role="link"
        tabindex="0"
        title="Help"
      >
        <svg
          height="2rem"
          viewBox="0 0 24 24"
          width="2rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 6a3.939 3.939 0 0 0-3.934 3.934h2C10.066 8.867 10.934 8 12
        8s1.934.867 1.934 1.934c0 .598-.481 1.032-1.216 1.626a9.208 9.208 0 0
        0-.691.599c-.998.997-1.027 2.056-1.027
        2.174V15h2l-.001-.633c.001-.016.033-.386.441-.793.15-.15.339-.3.535-.458.779-.631
        1.958-1.584 1.958-3.182A3.937 3.937 0 0 0 12 6zm-1 10h2v2h-2z"
          />
          <path
            d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12
        2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
          />
        </svg>
      </div>
       
      <a
        href="https://github.com/SavageCore/pd3-vault-cracker"
        target="_blank"
      >
        <div
          class="icon svelte-66di5t"
          title="View source"
        >
          <svg
            height="2rem"
            viewBox="0 0 24 24"
            width="2rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.026 2c-5.509 0-9.974 4.465-9.974 9.974 0 4.406 2.857 8.145 6.821
        9.465.499.09.679-.217.679-.481
        0-.237-.008-.865-.011-1.696-2.775.602-3.361-1.338-3.361-1.338-.452-1.152-1.107-1.459-1.107-1.459-.905-.619.069-.605.069-.605
        1.002.07 1.527 1.028 1.527 1.028.89 1.524 2.336 1.084
        2.902.829.091-.645.351-1.085.635-1.334-2.214-.251-4.542-1.107-4.542-4.93
        0-1.087.389-1.979 1.024-2.675-.101-.253-.446-1.268.099-2.64 0 0
        .837-.269 2.742 1.021a9.582 9.582 0 0 1 2.496-.336 9.554 9.554 0 0 1
        2.496.336c1.906-1.291 2.742-1.021 2.742-1.021.545 1.372.203 2.387.099
        2.64.64.696 1.024 1.587 1.024 2.675 0 3.833-2.33 4.675-4.552
        4.922.355.308.675.916.675 1.846 0 1.334-.012 2.41-.012 2.737 0
        .267.178.577.687.479C19.146 20.115 22 16.379 22 11.974 22 6.465 17.535 2
        12.026 2z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>
   
  <dialog>
    <article>
      <header
        class="svelte-11ga12t"
      >
        <a
          aria-label="Close"
          class="close"
          href="#close"
        />
         
        Help
      </header>
       
      <p
        class="svelte-11ga12t"
      >
        This tool helps you to find the correct combination for the vault in
      Payday 3.
      </p>
       
      <p
        class="svelte-11ga12t"
      >
        Enter the 2, 3 or 4 digits from the fingerprints on the vault keypad then
      click the
      
        <span
          class="green"
        >
          <svg
            height="1rem"
            viewBox="0 0 24 24"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m10 15.586-3.293-3.293-1.414 1.414L10 18.414l9.707-9.707-1.414-1.414z"
            />
          </svg>
        </span>
        
      button to calculate the possible combinations.
      </p>
       
      <p
        class="svelte-11ga12t"
      >
        To remove the entered digits click the
      
        <span
          class="red"
        >
          <svg
            height="1rem"
            viewBox="0 0 24 24"
            width="1rem"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m16.192 6.344-4.243 4.242-4.242-4.242-1.414 1.414L10.535 12l-4.242 4.242 1.414 1.414 4.242-4.242 4.243 4.242 1.414-1.414L13.364 12l4.242-4.242z"
            />
          </svg>
        </span>
        
      button. Right-click/long-press to clear all entered digits.
      </p>
       
      <p
        class="svelte-11ga12t"
      >
        Click on the 
        <svg
          height="1rem"
          viewBox="0 0 24 24"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.993 12c0 2.761 2.246 5.007 5.007 5.007s5.007-2.246 5.007-5.007S14.761 6.993 12 6.993 6.993 9.239 6.993 12zM12 8.993c1.658 0 3.007 1.349 3.007 3.007S13.658 15.007 12 15.007 8.993 13.658 8.993 12 10.342 8.993 12 8.993zM10.998 19h2v3h-2zm0-17h2v3h-2zm-9 9h3v2h-3zm17 0h3v2h-3zM4.219 18.363l2.12-2.122 1.415 1.414-2.12 2.122zM16.24 6.344l2.122-2.122 1.414 1.414-2.122 2.122zM6.342 7.759 4.22 5.637l1.415-1.414 2.12 2.122zm13.434 10.605-1.414 1.414-2.122-2.122 1.414-1.414z"
          />
        </svg>
         or
      
        <svg
          height="1rem"
          viewBox="0 0 24 24"
          width="1rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.742 13.045a8.088 8.088 0 0 1-2.077.271c-2.135 0-4.14-.83-5.646-2.336a8.025 8.025 0 0 1-2.064-7.723A1 1 0 0 0 9.73 2.034a10.014 10.014 0 0 0-4.489 2.582c-3.898 3.898-3.898 10.243 0 14.143a9.937 9.937 0 0 0 7.072 2.93 9.93 9.93 0 0 0 7.07-2.929 10.007 10.007 0 0 0 2.583-4.491 1.001 1.001 0 0 0-1.224-1.224zm-2.772 4.301a7.947 7.947 0 0 1-5.656 2.343 7.953 7.953 0 0 1-5.658-2.344c-3.118-3.119-3.118-8.195 0-11.314a7.923 7.923 0 0 1 2.06-1.483 10.027 10.027 0 0 0 2.89 7.848 9.972 9.972 0 0 0 7.848 2.891 8.036 8.036 0 0 1-1.484 2.059z"
          />
        </svg>
         icon to toggle between light and dark mode.
      </p>
       
      <h3
        class="svelte-11ga12t"
      >
        Keybinds
      </h3>
       
      <ol
        class="svelte-11ga12t"
      >
        <li>
          Enter/Return - Calculate combinations
        </li>
         
        <li>
          Backspace/Delete - Remove last entered digit
        </li>
         
        <li>
          Escape - Clear all entered digits
        </li>
         
        <li>
          Numbers or numpad - Enter digits from keyboard
        </li>
      </ol>
       
      <h3
        class="svelte-11ga12t"
      >
        Issues
      </h3>
       
      <p
        class="svelte-11ga12t"
      >
        Please post any issues on
      
        <a
          href="https://github.com/SavageCore/pd3-vault-cracker/issues"
          target="_blank"
        >
          GitHub
        </a>
        
      or DM
      
        <a
          href="https://www.reddit.com/message/compose/?to=SavageCore"
          target="_blank"
        >
          /u/SavageCore
        </a>
      </p>
    </article>
  </dialog>
</div>
`;
